modules/tappxBidAdapter.js

```diff
@@ -0,0 +1,284 @@
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+  */
+  isBidRequestValid: function(bid) {
+    if ((bid.params == null) || (bid.params.endpoint == null) || (bid.params.tappxkey == null)) {
+      utils.logWarn(`[TAPPX]: The Tappx endpoint and Tappxkey are mandatory. ${JSON.stringify(bid)}`);
+      return false;
+    }
+    return true;
+  },
+
+  /**
+   * Takes an array of valid bid requests, all of which are guaranteed to have passed the isBidRequestValid() test.
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {validBidRequests[]} - an array of bids
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    // Host definition
+    HOST = getParameters(validBidRequests, 'params.host');
+    hostDomain = HOST.split('/', 1)[0];
+
+    const ENDPOINT = getParameters(validBidRequests, 'params.endpoint');
+    const TAPPXKEY = getParameters(validBidRequests, 'params.tappxkey');
+    const BIDFLOOR = getParameters(validBidRequests, 'params.bidfloor');
+    const bannerMediaType = getParameters(validBidRequests, 'mediaTypes.banner');
+
+    let requests = [];
+    let payload = {};
+    let publisher = {};
+    let tagid;
+    let api = {};
+
+    // > App/Site object
+    if (getParameters(validBidRequests, 'params.app')) {
+      let app = {};
+      app.name = getParameters(validBidRequests, 'params.app.name');
+      app.bundle = getParameters(validBidRequests, 'params.app.bundle');
+      app.domain = getParameters(validBidRequests, 'params.app.domain');
+      publisher.name = getParameters(validBidRequests, 'params.app.publisher.name');
+      publisher.domain = getParameters(validBidRequests, 'params.app.publisher.domain');
+      tagid = `${app.name}_typeAdBanVid_${getOs()}`;
+      payload.app = app;
+      api[0] = getParameters(validBidRequests, 'params.api') ? getParameters(validBidRequests, 'params.api') : [3, 5];
+    } else {
+      let site = {};
+      site.name = window.location.hostname;
+      site.bundle = window.location.hostname;
+      site.domain = window.location.hostname;
+      publisher.name = window.location.hostname;
+      publisher.domain = window.location.hostname;
+      tagid = `${site.name}_typeAdBanVid_${getOs()}`;
+      payload.site = site;
+    }
+    // < App/Site object
+
+    // > Imp object
+    let imp = {};
+    let w;
+    let h;
+
+    if (bannerMediaType) {
+      let banner = {};
+      w = bannerMediaType.sizes[0][0];
+      h = bannerMediaType.sizes[0][1];
+      banner.w = w;
+      banner.h = h;
+      if ((bannerMediaType.sizes[0].indexOf(480) >= 0) && (bannerMediaType.sizes[0].indexOf(320) >= 0)) {
+        banner.pos = 7
+      } else {
+        banner.pos = 4
+      }
+
+      banner.api = api;
+
+      let format = {};
```

Comment:
do you really want this to be a hashmap or you meant array?

Reply:
Yes, we need a hashmap because we have a key value